src/controllers/hid/hidcontroller.cpp

```diff
@@ -243,13 +248,28 @@ int HidController::close() {
 bool HidController::poll() {
     Trace hidRead("HidController poll");
 
-    int result = hid_read(m_pHidDevice, m_pPollData, sizeof(m_pPollData) / sizeof(m_pPollData[0]));
-    if (result == -1) {
-        return false;
-    } else if (result > 0) {
-        Trace process("HidController process packet");
-        QByteArray outData(reinterpret_cast<char*>(m_pPollData), result);
-        receive(outData, mixxx::Time::elapsed());
+    int result = 1;
+    while (result > 0) {
+        // Rotate between two buffers so the memcmp below does not require deep copying to another buffer.
+        unsigned char* pCurrentBuffer = m_pPollData[m_iPollingBufferIndex];
+        m_iPollingBufferIndex = (m_iPollingBufferIndex + 1) % m_iNumBuffers;
+        unsigned char* pPreviousBuffer = m_pPollData[m_iPollingBufferIndex];
```

Comment:
Current and Previous buffers are swapped if you consider m_iNumBuffers > 2

Reply:
Oh, now that I take another look at this I think the naming would be clearer the other way around.

Reply:
fixed in cd283de